apps/src/lib/tools/jsdebugger/DebugConsole.jsx

```diff
@@ -211,11 +242,35 @@ export default connect(
 
     displayOutputToConsole() {
       if (this.props.logOutput.size > 0) {
-        return this.props.logOutput.map((output, i) => {
-          if (typeof output === 'object') {
-            output = output.toJS();
+        return this.props.logOutput.map((rowValue, i) => {
+          try {
+            rowValue = rowValue.toJS();
+          } catch (error) {
+            return <Inspector key={i} data={rowValue} />;
+          }
+          if (rowValue.input) {
+            return <div key={i}>&gt; {rowValue.input}</div>;
+          } else if (
+            rowValue.output ||
+            FALSY_VALUES.has(rowValue.output) ||
+            (rowValue.output === undefined && rowValue.fromConsoleLog) ||
+            (rowValue.output === undefined &&
+              !rowValue.fromConsoleLog &&
+              rowValue.undefinedInput)
```

Comment:
NIT: Replace 256-259 with 
rowValue.output === undefined && (rowValue.fromConsoleLog) || rowValue.undefinedInput)

Reply:
Awesome! I'll refactor it to that.